cohortextractor/study_definition.py

```diff
@@ -280,11 +281,49 @@ def add_day_to_date(val):
     def make_df_from_expectations(self, population):
         df = pd.DataFrame()
 
-        # Start with dates, so we can use them as inputs for incidence
-        # matching on dependent columns
+        # Start with the user-supplied CSV file
+        for colname, definition_args in self.pandas_csv_args["args"].items():
+            if definition_args["funcname"] not in [
+                "with_value_from_file",
+                "which_exist_in_file",
+            ]:
+                continue
+            f_path = definition_args["f_path"]
+            if not is_csv_filename(f_path):
+                raise ValueError("`f_path` must be a path to a CSV file")
+
+            returning = definition_args.get("returning")
+            returning_type = definition_args.get("returning_type")
+            if returning is not None and returning_type is None:
+                raise ValueError(f"No `returning_type` defined for {colname}")
+            if returning is None and returning_type is not None:
+                raise ValueError(f"No `returning` defined for {colname}")
+            if returning is not None and not re.fullmatch(r"\w+", returning, re.ASCII):
+                raise ValueError(
+                    f"{colname} should contain only alphanumeric characters and the underscore character"
```

Comment:
These checks duplicate what the actual functions do; is it worth pulling them out to a shared utility somewhere?

Reply:
I wasn't sure, to be honest. In the actual functions, I've raised `TypeError`; within the `study_definition` module, `ValueError` is the norm. (I mentioned being unsure about what type of error to raise during today's meeting.) In a couple of cases, the actual functions don't have access to the column names, so the error messages can't be as specific (and hence, useful). For these reasons, I didn't pull the checks into a shared utility.